grafana/generic/configure-grafana-datasource/tasks/main.yml

```diff
@@ -1,17 +1,35 @@
 ---
-- name: create datasource
+- name: create datasource prometheus
   uri:
     url: "http://{{ ansible_host }}:{{ grafana_port }}/api/datasources"
     method: POST
-    body: "{{ lookup ('template', 'datasource.json.j2') | to_json }}"
+    body: "{{ lookup ('template', 'datasource_prometheus.json.j2') | to_json }}"
     body_format: json
     url_username: "{{ grafana_user }}"
     url_password: "{{ grafana_password }}"
     force_basic_auth: true
     headers:
       Accept: "application/json"
       Content-type: "application/json"
-  loop: "{{ groups['grafana_datasources'] }}"
+  loop: "{{ groups['grafana_prometheus_datasources'] }}"
+  register: result
+  failed_when:
+    - '"already exists" not in result.json.message'
+    - result.status != 200
```

Comment:
I think you might want to use the `status_code` parameter to the `uri` module instead and allow for both `[200,409]`. That should let you remove the `failed_when`.

Reply:
Fixed. Thanks for pointing this out to me!